packages/camera/camera_android_camerax/lib/src/aspect_ratio_strategy.dart

```diff
@@ -0,0 +1,138 @@
+import 'package:meta/meta.dart' show immutable;
+
+import 'camerax_library.g.dart';
+import 'instance_manager.dart';
+import 'java_object.dart';
+
+/// The aspect ratio of a UseCase.
+///
+/// Aspect ratio is the ratio of width to height.
+///
+/// See https://developer.android.com/reference/androidx/camera/core/AspectRatio.
+class AspectRatio {
+  AspectRatio._();
+
+  /// 4:3 standard aspect ratio.
+  ///
+  /// See https://developer.android.com/reference/androidx/camera/core/AspectRatio#RATIO_4_3().
+  static const int ratio4To3 = 0;
+
+  /// 16:9 standard aspect ratio.
+  ///
+  /// See https://developer.android.com/reference/androidx/camera/core/AspectRatio#RATIO_16_9().
+  static const int ratio16To9 = 1;
+
+  /// The aspect ratio representing no preference for aspect ratio.
+  ///
+  /// See https://developer.android.com/reference/androidx/camera/core/AspectRatio#RATIO_DEFAULT().
+  static const int ratioDefault = -1;
+}
+
+/// The aspect ratio strategy defines the sequence of aspect ratios that are
+/// used to select the best size for a particular image.
+///
+/// See https://developer.android.com/reference/androidx/camera/core/resolutionselector/AspectRatioStrategy.
+@immutable
+class AspectRatioStrategy extends JavaObject {
+  /// Construct a [AspectRatioStrategy].
+  AspectRatioStrategy({
+    required this.preferredAspectRatio,
+    required this.fallbackRule,
+    super.binaryMessenger,
+    super.instanceManager,
+  })  : _api = _AspectRatioStrategyHostApiImpl(
+          instanceManager: instanceManager,
+          binaryMessenger: binaryMessenger,
+        ),
+        super.detached() {
+    _api.createFromInstances(this, preferredAspectRatio, fallbackRule);
+  }
+
+  /// Instantiates a [AspectRatioStrategy] without creating and attaching to an
+  /// instance of the associated native class.
+  ///
+  /// This should only be used outside of tests by subclasses created by this
+  /// library or to create a copy for an [InstanceManager].
+  AspectRatioStrategy.detached({
+    required this.preferredAspectRatio,
+    required this.fallbackRule,
+    super.binaryMessenger,
+    super.instanceManager,
+  })  : _api = _AspectRatioStrategyHostApiImpl(
+          instanceManager: instanceManager,
+          binaryMessenger: binaryMessenger,
+        ),
+        super.detached();
+
+  /// CameraX doesn't fall back to select sizes of any other aspect ratio when
+  /// this fallback rule is used.
+  ///
+  /// See https://developer.android.com/reference/androidx/camera/core/resolutionselector/AspectRatioStrategy#FALLBACK_RULE_NONE().
+  static const int fallbackRuleNone = 0;
+
+  /// CameraX automatically chooses the next best aspect ratio which contains
+  /// the closest field of view (FOV) of the camera sensor, from the remaining
+  /// options.
+  ///
+  /// See https://developer.android.com/reference/androidx/camera/core/resolutionselector/AspectRatioStrategy#FALLBACK_RULE_AUTO().
+  static const int fallbackRuleAuto = 1;
```

Comment:
Consider bumping these constants to the top of the class for easier discoverability.

Reply:
This would go against the pattern I've been following but if you have strong opinions I can follow this up with another PR moving all static constants to the top of their respective classes